File: java/ListNode.java
 public class ListNode {
   int val;
   ListNode next;

   ListNode(int x) {
		 val = x;
	 }

	 public static ListNode create(int[] A) {
		 ListNode head = new ListNode(0);
		 ListNode p = head;

		 for (int n : A) {
			 p.next = new ListNode(n);
			 p = p.next;
		 }

		 return head.next;
	 }

	 public static void print(ListNode head) {
		 System.out.print("[ ");
		 while (head != null) {
			 System.out.print(head.val + " ");
			 head = head.next;
		 }
		 System.out.println("]");
	 }
 }

File: listNode.cpp
/*
 * Filename: listNode.cpp
 * Author: Jun Xu
 * Email: junx@cs.bu.edu
 * Created Time: Sat 26 Dec 2015 09:47:47 PM EST
 * Description: 
 */

#ifndef LIST_NODE_H
#define LIST_NODE_H

struct ListNode
{
	int val;
	ListNode *next;
	ListNode(int x) : val(x), next(NULL) {}
};

#endif
